diff --git a/.idea/.gitignore b/.idea/.gitignore
new file mode 100644
index 0000000000000000000000000000000000000000..b58b603fea78041071d125a30db58d79b3d49217
--- /dev/null
+++ b/.idea/.gitignore
@@ -0,0 +1,5 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
diff --git a/.idea/modules.xml b/.idea/modules.xml
new file mode 100644
index 0000000000000000000000000000000000000000..d9fb14e4d06a030cba3cd105982c4f196e0718a0
--- /dev/null
+++ b/.idea/modules.xml
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/user.iml" filepath="$PROJECT_DIR$/.idea/user.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
diff --git a/.idea/user.iml b/.idea/user.iml
new file mode 100644
index 0000000000000000000000000000000000000000..24643cc37449b4bde54411a80b8ed61258225e34
--- /dev/null
+++ b/.idea/user.iml
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/.tmp" />
+      <excludeFolder url="file://$MODULE_DIR$/temp" />
+      <excludeFolder url="file://$MODULE_DIR$/tmp" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
diff --git a/.idea/workspace.xml b/.idea/workspace.xml
new file mode 100644
index 0000000000000000000000000000000000000000..b85767580607b78ac1b3b7d223a55e880964d0e5
--- /dev/null
+++ b/.idea/workspace.xml
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ChangeListManager">
+    <list default="true" id="96d23922-9b26-410b-9f23-f3c8a7aeadde" name="Changes" comment="" />
+    <option name="SHOW_DIALOG" value="false" />
+    <option name="HIGHLIGHT_CONFLICTS" value="true" />
+    <option name="HIGHLIGHT_NON_ACTIVE_CHANGELIST" value="false" />
+    <option name="LAST_RESOLUTION" value="IGNORE" />
+  </component>
+  <component name="ProjectColorInfo"><![CDATA[{
+  "associatedIndex": 1
+}]]></component>
+  <component name="ProjectId" id="2lLeoNSjwkY3AMMtx4iM8cWcKLE" />
+  <component name="ProjectViewState">
+    <option name="hideEmptyMiddlePackages" value="true" />
+    <option name="showExcludedFiles" value="false" />
+    <option name="showLibraryContents" value="true" />
+  </component>
+  <component name="PropertiesComponent"><![CDATA[{
+  "keyToString": {
+    "RunOnceActivity.ShowReadmeOnStart": "true",
+    "nodejs_package_manager_path": "npm",
+    "vue.rearranger.settings.migration": "true"
+  }
+}]]></component>
+  <component name="SharedIndexes">
+    <attachedChunks>
+      <set>
+        <option value="bundled-js-predefined-1d06a55b98c1-2e7d6887c066-JavaScript-WS-241.15989.47" />
+      </set>
+    </attachedChunks>
+  </component>
+  <component name="SpellCheckerSettings" RuntimeDictionaries="0" Folders="0" CustomDictionaries="0" DefaultDictionary="application-level" UseSingleDictionary="true" transferred="true" />
+  <component name="TaskManager">
+    <task active="true" id="Default" summary="Default task">
+      <changelist id="96d23922-9b26-410b-9f23-f3c8a7aeadde" name="Changes" comment="" />
+      <created>1724963957894</created>
+      <option name="number" value="Default" />
+      <option name="presentableId" value="Default" />
+      <updated>1724963957894</updated>
+      <workItem from="1724963959003" duration="23000" />
+    </task>
+    <servers />
+  </component>
+  <component name="TypeScriptGeneratedFilesManager">
+    <option name="version" value="3" />
+  </component>
+</project>
\ No newline at end of file
diff --git a/commonjs/generated/google.cjs b/commonjs/generated/google.cjs
index 2332b9869ee501899c9c494b97e18860e78b0ad3..0bcd27f3807e45fdeaae74fbb7cc4fdbc735aa6d 100644
--- a/commonjs/generated/google.cjs
+++ b/commonjs/generated/google.cjs
@@ -1,9 +1,15 @@
-const googleFileList = [
-  '../../out/google/api/health_check/v1/health_check_grpc_pb.js',
-  '../../out/google/api/health_check/v1/health_check_pb.js',
-]
+//const googleFileList = [
+//  '../../out/google/api/health_check/v1/health_check_grpc_pb.js',
+//  '../../out/google/api/health_check/v1/health_check_pb.js',
+//]
+
+//module.exports = googleFileList.reduce((acc, pkg) => ({
+//  ...acc,
+//  ...require(pkg),
+//}), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
+
+module.exports = {
+  ...require('../../out/google/api/health_check/v1/health_check_grpc_pb.js'),
+  ...require('../../out/google/api/health_check/v1/health_check_pb.js'),
+}
 
-module.exports = googleFileList.reduce((acc, pkg) => ({
-  ...acc,
-  ...require(pkg),
-}), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
diff --git a/commonjs/generated/puppet.cjs b/commonjs/generated/puppet.cjs
index 9fadd30e402c05c5bf70f0fa34eab45a25545021..3e7449ef25487ab1780cefc084ac60e7a79a23dd 100644
--- a/commonjs/generated/puppet.cjs
+++ b/commonjs/generated/puppet.cjs
@@ -1,39 +1,59 @@
-const puppetFileList = [
-  '../../out/wechaty/deprecated/file-box_pb.js',
+// const puppetFileList = [
+//   '../../out/wechaty/deprecated/file-box_pb.js',
+//
+//   '../../out/wechaty/puppet/base_pb.js',
+//   '../../out/wechaty/puppet/contact_pb.js',
+//   '../../out/wechaty/puppet/download-upload_pb.js',
+//   '../../out/wechaty/puppet/event_pb.js',
+//   '../../out/wechaty/puppet/friendship_pb.js',
+//   '../../out/wechaty/puppet/location_pb.js',
+//   '../../out/wechaty/puppet/message_pb.js',
+//   '../../out/wechaty/puppet/mini-program_pb.js',
+//   '../../out/wechaty/puppet/referrer_pb.js',
+//   '../../out/wechaty/puppet/room_pb.js',
+//   '../../out/wechaty/puppet/room-invitation_pb.js',
+//   '../../out/wechaty/puppet/room-member_pb.js',
+//   '../../out/wechaty/puppet/tag_pb.js',
+//   '../../out/wechaty/puppet/url-link_pb.js',
+//
+//   '../../out/wechaty/puppet_grpc_pb.js',
+//   '../../out/wechaty/puppet_pb.js',
+// ]
+//
+// /**
+//  * Huan(202108):
+//  *  if there's a "package.json" file in the `out/` directory,
+//  *    then all the files in the `out/` directory will be treated as one module,
+//  *    which means tht `require` each file under that directory will add methods to the same module.
+//  */
+// // for (const pkg of pkgs) {
+// //   console.info('## pkg:', pkg)
+// //   const module = require(pkg)
+// //   console.info(Object.keys(module).length)
+// //   // OOPS! The output number above will be keep increasing
+// // }
+//
+// module.exports = puppetFileList.reduce((acc, pkg) => ({
+//   ...acc,
+//   ...require(pkg),
+// }), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
 
-  '../../out/wechaty/puppet/base_pb.js',
-  '../../out/wechaty/puppet/contact_pb.js',
-  '../../out/wechaty/puppet/download-upload_pb.js',
-  '../../out/wechaty/puppet/event_pb.js',
-  '../../out/wechaty/puppet/friendship_pb.js',
-  '../../out/wechaty/puppet/location_pb.js',
-  '../../out/wechaty/puppet/message_pb.js',
-  '../../out/wechaty/puppet/mini-program_pb.js',
-  '../../out/wechaty/puppet/referrer_pb.js',
-  '../../out/wechaty/puppet/room_pb.js',
-  '../../out/wechaty/puppet/room-invitation_pb.js',
-  '../../out/wechaty/puppet/room-member_pb.js',
-  '../../out/wechaty/puppet/tag_pb.js',
-  '../../out/wechaty/puppet/url-link_pb.js',
-
-  '../../out/wechaty/puppet_grpc_pb.js',
-  '../../out/wechaty/puppet_pb.js',
-]
-
-/**
- * Huan(202108):
- *  if there's a "package.json" file in the `out/` directory,
- *    then all the files in the `out/` directory will be treated as one module,
- *    which means tht `require` each file under that directory will add methods to the same module.
- */
-// for (const pkg of pkgs) {
-//   console.info('## pkg:', pkg)
-//   const module = require(pkg)
-//   console.info(Object.keys(module).length)
-//   // OOPS! The output number above will be keep increasing
-// }
-
-module.exports = puppetFileList.reduce((acc, pkg) => ({
-  ...acc,
-  ...require(pkg),
-}), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
+module.exports = {
+  ...require('../../out/wechaty/deprecated/file-box_pb.js'),
+  ...require('../../out/wechaty/puppet/base_pb.js'),
+  ...require('../../out/wechaty/puppet/contact_pb.js'),
+  ...require('../../out/wechaty/puppet/download-upload_pb.js'),
+  ...require('../../out/wechaty/puppet/event_pb.js'),
+  ...require('../../out/wechaty/puppet/friendship_pb.js'),
+  ...require('../../out/wechaty/puppet/location_pb.js'),
+  ...require('../../out/wechaty/puppet/message_pb.js'),
+  ...require('../../out/wechaty/puppet/mini-program_pb.js'),
+  ...require('../../out/wechaty/puppet/referrer_pb.js'),
+  ...require('../../out/wechaty/puppet/room_pb.js'),
+  ...require('../../out/wechaty/puppet/room-invitation_pb.js'),
+  ...require('../../out/wechaty/puppet/room-member_pb.js'),
+  ...require('../../out/wechaty/puppet/tag_pb.js'),
+  ...require('../../out/wechaty/puppet/url-link_pb.js'),
+  ...require('../../out/wechaty/puppet_grpc_pb.js'),
+  ...require('../../out/wechaty/puppet_pb.js'),
+}
