diff --git a/.idea/.gitignore b/.idea/.gitignore
new file mode 100644
index 0000000000000000000000000000000000000000..b58b603fea78041071d125a30db58d79b3d49217
--- /dev/null
+++ b/.idea/.gitignore
@@ -0,0 +1,5 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
diff --git a/.idea/workspace.xml b/.idea/workspace.xml
new file mode 100644
index 0000000000000000000000000000000000000000..8744590a49ab577e0ec11e1cf8baa9c658ba2719
--- /dev/null
+++ b/.idea/workspace.xml
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectViewState">
+    <option name="hideEmptyMiddlePackages" value="true" />
+    <option name="showExcludedFiles" value="false" />
+    <option name="showLibraryContents" value="true" />
+  </component>
+  <component name="PropertiesComponent">{}</component>
+</project>
\ No newline at end of file
diff --git a/dist/esm/commonjs/generated/google.cjs b/dist/esm/commonjs/generated/google.cjs
index 2332b9869ee501899c9c494b97e18860e78b0ad3..2d4367ed341f136249b3985ffe3bda38bca6d42d 100644
--- a/dist/esm/commonjs/generated/google.cjs
+++ b/dist/esm/commonjs/generated/google.cjs
@@ -1,9 +1,15 @@
-const googleFileList = [
-  '../../out/google/api/health_check/v1/health_check_grpc_pb.js',
-  '../../out/google/api/health_check/v1/health_check_pb.js',
-]
+// const googleFileList = [
+//   '../../out/google/api/health_check/v1/health_check_grpc_pb.js',
+//   '../../out/google/api/health_check/v1/health_check_pb.js',
+// ]
 
-module.exports = googleFileList.reduce((acc, pkg) => ({
-  ...acc,
-  ...require(pkg),
-}), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
+// module.exports = googleFileList.reduce((acc, pkg) => ({
+//   ...acc,
+//   ...require(pkg),
+// }), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
+
+// markshawn2020@2024-08-26
+module.exports = {
+  ...require('../../out/google/api/health_check/v1/health_check_grpc_pb.js'),
+  ...require('../../out/google/api/health_check/v1/health_check_pb.js'),
+}
\ No newline at end of file
diff --git a/dist/esm/commonjs/generated/puppet.cjs b/dist/esm/commonjs/generated/puppet.cjs
index 9fadd30e402c05c5bf70f0fa34eab45a25545021..e690c4dc3713e54df802a6666e3ff2bc62bf0730 100644
--- a/dist/esm/commonjs/generated/puppet.cjs
+++ b/dist/esm/commonjs/generated/puppet.cjs
@@ -1,24 +1,24 @@
-const puppetFileList = [
-  '../../out/wechaty/deprecated/file-box_pb.js',
-
-  '../../out/wechaty/puppet/base_pb.js',
-  '../../out/wechaty/puppet/contact_pb.js',
-  '../../out/wechaty/puppet/download-upload_pb.js',
-  '../../out/wechaty/puppet/event_pb.js',
-  '../../out/wechaty/puppet/friendship_pb.js',
-  '../../out/wechaty/puppet/location_pb.js',
-  '../../out/wechaty/puppet/message_pb.js',
-  '../../out/wechaty/puppet/mini-program_pb.js',
-  '../../out/wechaty/puppet/referrer_pb.js',
-  '../../out/wechaty/puppet/room_pb.js',
-  '../../out/wechaty/puppet/room-invitation_pb.js',
-  '../../out/wechaty/puppet/room-member_pb.js',
-  '../../out/wechaty/puppet/tag_pb.js',
-  '../../out/wechaty/puppet/url-link_pb.js',
-
-  '../../out/wechaty/puppet_grpc_pb.js',
-  '../../out/wechaty/puppet_pb.js',
-]
+// const puppetFileList = [
+//   '../../out/wechaty/deprecated/file-box_pb.js',
+//
+//   '../../out/wechaty/puppet/base_pb.js',
+//   '../../out/wechaty/puppet/contact_pb.js',
+//   '../../out/wechaty/puppet/download-upload_pb.js',
+//   '../../out/wechaty/puppet/event_pb.js',
+//   '../../out/wechaty/puppet/friendship_pb.js',
+//   '../../out/wechaty/puppet/location_pb.js',
+//   '../../out/wechaty/puppet/message_pb.js',
+//   '../../out/wechaty/puppet/mini-program_pb.js',
+//   '../../out/wechaty/puppet/referrer_pb.js',
+//   '../../out/wechaty/puppet/room_pb.js',
+//   '../../out/wechaty/puppet/room-invitation_pb.js',
+//   '../../out/wechaty/puppet/room-member_pb.js',
+//   '../../out/wechaty/puppet/tag_pb.js',
+//   '../../out/wechaty/puppet/url-link_pb.js',
+//
+//   '../../out/wechaty/puppet_grpc_pb.js',
+//   '../../out/wechaty/puppet_pb.js',
+// ]
 
 /**
  * Huan(202108):
@@ -33,7 +33,28 @@ const puppetFileList = [
 //   // OOPS! The output number above will be keep increasing
 // }
 
-module.exports = puppetFileList.reduce((acc, pkg) => ({
-  ...acc,
-  ...require(pkg),
-}), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
+// module.exports = puppetFileList.reduce((acc, pkg) => ({
+//   ...acc,
+//   ...require(pkg),
+// }), {}) // Huan(202108): must provide a `{}` as the initial value, or it will be `[]`
+
+// markshawn2020@2024-08-26
+module.exports = {
+  ...require('../../out/wechaty/deprecated/file-box_pb.js'),
+  ...require('../../out/wechaty/puppet/base_pb.js'),
+  ...require('../../out/wechaty/puppet/contact_pb.js'),
+  ...require('../../out/wechaty/puppet/download-upload_pb.js'),
+  ...require('../../out/wechaty/puppet/event_pb.js'),
+  ...require('../../out/wechaty/puppet/friendship_pb.js'),
+  ...require('../../out/wechaty/puppet/location_pb.js'),
+  ...require('../../out/wechaty/puppet/message_pb.js'),
+  ...require('../../out/wechaty/puppet/mini-program_pb.js'),
+  ...require('../../out/wechaty/puppet/referrer_pb.js'),
+  ...require('../../out/wechaty/puppet/room_pb.js'),
+  ...require('../../out/wechaty/puppet/room-invitation_pb.js'),
+  ...require('../../out/wechaty/puppet/room-member_pb.js'),
+  ...require('../../out/wechaty/puppet/tag_pb.js'),
+  ...require('../../out/wechaty/puppet/url-link_pb.js'),
+  ...require('../../out/wechaty/puppet_grpc_pb.js'),
+  ...require('../../out/wechaty/puppet_pb.js'),
+}
\ No newline at end of file
